File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.py
import sys

string = list(sys.stdin.readline().strip())
# '<'시작하는지 판단해야함.
# '<'로 시작했으면 '>'까지 temp에 임시 저장
# '>' 만났으면 temp에 있는 거 거꾸로 뒤집에서 결과배열에 추가. 이후 초기화
# '<'를 만났는지 판단하기 위한 check boolean 추가

check = False
temp = []
result = []

for i in string:

    if check == True:
        if i == '>':
            check = False
            for j in temp:
                result.append(j)
            result.append(i)
            temp.clear()
        else:
            temp.append(i)
    else:
        if i == '<':
            if len(temp) != 0:
                temp.reverse()
                for j in temp:
                    result.append(j)
                temp.clear()
            check = True
            result.append(i)
        if check == False and i == ' ':
            temp.reverse()
            for j in temp:
                result.append(j)
            temp.clear()
            result.append(i)
        elif check == False:
            temp.append(i)
    
    
temp.reverse()
for j in temp:
    result.append(j)

for i in result:
    print(i, end="")

File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.java
import javax.swing.*;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        String s = bf.readLine();

        StringBuilder mainSb = new StringBuilder();
        StringBuilder subSb = new StringBuilder();

        boolean check = false;

        for(int i=0; i<s.length(); i++){
            if((s.charAt(i) >= 'a' && s.charAt(i) <= 'z') || (s.charAt(i) >= '0' && s.charAt(i) <= '9')){
                if(check) mainSb.append(s.charAt(i));
                else subSb.append(s.charAt(i));
            }else if(s.charAt(i) == '<' || s.charAt(i) == '>'){
                if(subSb.length() != 0){
                    mainSb.append(subSb.reverse());
                    subSb.setLength(0);
                }
                if(s.charAt(i) == '<') check = true;
                else check = false;
                mainSb.append(s.charAt(i));
            }else if(s.charAt(i) == ' '){
                if(check) mainSb.append(' ');
                else {
                    subSb.reverse();
                    mainSb.append(subSb).append(' ');
                    subSb.setLength(0);
                }
            }
        }
        if(subSb.length() != 0){
            mainSb.append(subSb.reverse());
        }

        System.out.println(mainSb);
    }


}


